scripts/install-build-tools.sh

```diff
@@ -6,19 +6,61 @@ green="\033[0;32m"
 cyan="\033[0;36m"
 end="\033[0m"
 
-set -e
+# for debugging, 'set -x' can be added to trace the commands
+set -ex
 
 SUDO=''
 if (( $EUID != 0 )); then
     echo -e "non-root user, sudo required"
     SUDO='sudo'
 fi
 
+# install in a custom prefix rather than /usr/local. by default, this
+# chooses "prefix" directory alongside "scripts" directory.
+PREFIX="$(cd "$(dirname "$0")"/.. && pwd)/prefix"
+echo "Will install local dependencies in the following prefix: $PREFIX"
+mkdir -p $PREFIX $PREFIX/bin $PREFIX/lib $PREFIX/include
```

Comment:
```suggestion
mkdir -p "$PREFIX"/{bin,lib,include}
```

Reply:
That's clean, I'll add to `setup-dependencies.sh` as well